orc/src/main/java/org/apache/iceberg/orc/OrcValueReaders.java

```diff
@@ -181,12 +173,25 @@ private T readInternal(T struct, ColumnVector[] columnVectors, int row) {
       for (int c = 0; c < readers.length; ++c) {
         set(struct, c, reader(c).read(columnVectors[c], row));
       }
+      return struct;
+    }
+  }
 
-      for (int i = 0; i < positions.length; i += 1) {
-        set(struct, positions[i], constants[i]);
-      }
+  private static class ConstantReader<C> implements OrcValueReader<C> {
+    private final C constant;
 
-      return struct;
+    private ConstantReader(C constant) {
+      this.constant = constant;
+    }
+
+    @Override
+    public C read(ColumnVector ignored, int ignoredRow) {
```

Comment:
Is `ColumnVector` still materialized? Is it possible to avoid reading that entirely?

Reply:
Yes, I guess we can do that by not asking ORC to project these column. Let me give it a try.

Reply:
That's going to be a bigger time savings and that's what we do for Parquet. We just drop it from the projection we pass down to the format.